protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SetAnnotationCommandParserTest.java

```diff
@@ -0,0 +1,88 @@
+    private static final MailboxAnnotation SHARED_ANNOTATION = MailboxAnnotation.newInstance("/shared/comment", "This one is for you!");
+    private static final MailboxAnnotation NIL_ANNOTATION = MailboxAnnotation.nil("/private/comment");
+    private SetAnnotationCommandParser parser = new SetAnnotationCommandParser();
+    private ImapCommand command = ImapCommand.anyStateCommand("Command");
+
+    @Test
+    public void decodeMessageShouldReturnRequestContainsOneAnnotation() throws DecodingException {
+        SetAnnotationRequest request = createRequestFromStringCommand("INBOX (/private/comment \"This is my comment\") \n", null);
+
+        assertThat(request.getMailboxName()).isEqualTo(INBOX);
+        assertThat(request.getMailboxAnnotation()).containsOnly(PRIVATE_ANNOTATION);
+    }
+
+    @Test(expected = DecodingException.class)
+    public void decodeMessageShouldThrowDecodingExceptionWhenContainsInvalidAnnotationKey() throws DecodingException {
+        createRequestFromStringCommand("INBOX (/pri*vate/comment \"This is my comment\") \n", null);
+    }
+
+    @Test
+    public void decodeMessageShouldReturnRequestContainsOneNilAnnotation() throws DecodingException {
+        SetAnnotationRequest request = createRequestFromStringCommand("INBOX (/private/comment NIL) \n", null);
+
+        assertThat(request.getMailboxName()).isEqualTo(INBOX);
+        assertThat(request.getMailboxAnnotation()).containsOnly(NIL_ANNOTATION);
+    }
+
+    @Test
+    public void decodeMessageShouldReturnRequestContainsOneAnnotationWithMultiLinesValue() throws DecodingException {
+        SetAnnotationRequest request = createRequestFromStringCommand("INBOX (/private/comment {32}\nMy new comment across two lines.) \n", new ByteArrayOutputStream());
+
+        assertThat(request.getMailboxName()).isEqualTo(INBOX);
+        assertThat(request.getMailboxAnnotation()).containsOnly(MailboxAnnotation.newInstance("/private/comment", "My new comment across two lines."));
+    }
+    @Test
+    public void decodeMessageShouldReturnRequestContainsMultiAnnotations() throws DecodingException {
+        SetAnnotationRequest request = createRequestFromStringCommand("INBOX (/private/comment \"This is my comment\" /shared/comment \"This one is for you!\") \n", null);
+
+        assertThat(request.getMailboxName()).isEqualTo(INBOX);
+        assertThat(request.getMailboxAnnotation()).containsExactly(PRIVATE_ANNOTATION, SHARED_ANNOTATION);
+    }
+
+    private SetAnnotationRequest createRequestFromStringCommand(String commandString, OutputStream output) throws DecodingException {
+        InputStream inputStream = new ByteArrayInputStream(commandString.getBytes());
+        ImapRequestStreamLineReader lineReader = new ImapRequestStreamLineReader(inputStream, output);
+
+        return (SetAnnotationRequest) parser.decode(command, lineReader, "a", null);
+    }
```

Comment:
you lack a lot of edge cases : unbalanced parenthesis, invalid quoted strings, too many arguments, too few, etc